index.js

```diff
@@ -62,6 +62,39 @@ class ScmBase {
         return Promise.reject(new Error('Not implemented'));
     }
 
+    /**
+     * Checks whether auto deploy key generation is switched on or not
+     * @method checkAutoDeployKeyGeneration
+     * @param  {Object}    config                   Configuration
+     * @param  {String}    [config.scmContext]      The scm context name
+     * @return {Promise}
+     */
+    checkAutoDeployKeyGeneration(config) {
```

Comment:
I think it might be better to rename this to something that asks a 'yes or no' question
maybe prefer something like `autoDeployKeyGenerationEnabled`? not sure what would be better
just so it's more obvious it's a boolean

Reply:
yeah I too think `autoDeployKeyGenerationEnabled` is more descriptive for the purpose. I'll change it.